pkg/scheduler/framework/plugins/volumezone/volume_zone.go

```diff
@@ -42,28 +41,121 @@ type VolumeZone struct {
 }
 
 var _ framework.FilterPlugin = &VolumeZone{}
+var _ framework.PreFilterPlugin = &VolumeZone{}
 var _ framework.EnqueueExtensions = &VolumeZone{}
 
 const (
 	// Name is the name of the plugin used in the plugin registry and configurations.
 	Name = names.VolumeZone
 
+	stateKey framework.StateKey = Name
+
 	// ErrReasonConflict is used for NoVolumeZoneConflict predicate error.
 	ErrReasonConflict = "node(s) had no available volume zone"
 )
 
-var volumeZoneLabels = sets.NewString(
+// the state is initialized in PreFilter phase. because we save the pointer in
+// framework.CycleState, in the later phases we don't need to call Write method
+// to update the value
+type stateData struct {
+	// pvByVolume holds the pv information corresponding to pod's volume
+	// it's initialized in the PreFilter phase
+	pvByVolume []*v1.PersistentVolume
```

Comment:
```suggestion
	podPVs []*v1.PersistentVolume
```

Reply:
Or even

```suggestion
	podPVTopologies [][]pvTopology
```

where `pvTopology` holds the name of the topology label and the set of values. I don't think we need any other information from the PV.

Reply:
We need pv's name to log related messages, so maybe `pvTopology` also needs to hold a pv's name? @alculquicondor 
```
if !volumeVSet.Has(nodeV) {
					klog.V(10).InfoS("Won't schedule pod onto node due to volume (mismatch on label key)", "pod", klog.KObj(pod), "node", klog.KObj(node), "PV", klog.KRef("", pv.Name), "PVLabelKey", labelName)
					return framework.NewStatus(framework.UnschedulableAndUnresolvable, ErrReasonConflict)
				}
```


> where `pvTopology` holds the name of the topology label and the set of values. I don't think we need any other information from the PV.

Reply:
Yes, add what's necessary. The key point is that we shouldn't repeat calculations in Filter. Currently we are parsing the set of values over and over